dipy/denoise/denspeed.pyx

```diff
@@ -108,9 +103,10 @@ def _nlmeans_3d(double[:, :, ::1] arr, double[:, :, ::1] mask,
 
     # move the block
     with nogil, parallel():
-        for i in prange(B + P, I - B - P, schedule="dynamic"):
```

Comment:
Yeah identical, the speed improvements are from 4.4 seconds to 3.7 seconds with 4 threads